### PR TITLE
fourier_filter_example

### PR DESCRIPTION
**Pull request by erinlincoln**
- fourier_filter function: https://github.com/NCAR/geocat-comp

image
image
image

### CODE REVIEW

**Comment by michaelavs** [comment]
@pilotchute @anissa111 When running this one locally (and attempting to build other examples with sphinx) there is a "ModuleNotFound" error from climatology.py for cf_xarray. Is this an actual naming error or is cf_xarray another package that should be added to the geocat-example conda environment?

**Review by anissa111 — COMMENTED** [review]
> @pilotchute @anissa111 When running this one locally (and attempting to build other examples with sphinx) there is a "ModuleNotFound" error from climatology.py for cf_xarray. Is this an actual naming error or is cf_xarray another package that should be added to the geocat-example conda environment?
> 
> Edit: actually it looks like cf_xarray isn't used in climatology after being imported, should we get rid of it in climatology?

@michaelavs Actually no! Anderson and Orhan taught me this on a recent -comp PR. You need to import cf_xarray even though the IDE will say it's not being used. I'm not getting that error locally, though.

File: GeoCAT-comp-examples/matplotlib/fourier_filter_example.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                         band_block=True)
+ax.plot(no_tide[2000:3000])
+
+# Show figure
+fig.show()
+
+# Generate figure with 2 by 1 subplots and set its size (width, height) in inches
+fig, axs = plt.subplots(2, 1, dpi=100, figsize=(8, 4), constrained_layout=True)
+
+# Plot the real set of data utilizing NumPy's Fouriet Transform function using both
```
Comment (anissa111) [inline GeoCAT-comp-examples/matplotlib/fourier_filter_example.py:87]:
```suggestion
# Plot the real set of data utilizing NumPy's Fourier Transform function using both
```

File: GeoCAT-comp-examples/matplotlib/fourier_filter_example.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+# the original data and the fourier_filter applied to the second set of cutoffs
+axs[0].set_title('real')
+axs[0].plot(np.real(np.fft.fft(xr_data)[1:100]))
+axs[0].plot(np.real(np.fft.fft(no_tide)[1:100]))
+
+# Plot the imaginary set of data utilizing NumPy's Fouriet Transform function using both
```
Comment (anissa111) [inline GeoCAT-comp-examples/matplotlib/fourier_filter_example.py:93]:
```suggestion
# Plot the imaginary set of data utilizing NumPy's Fourier Transform function using both
```

File: GeoCAT-comp-examples/matplotlib/fourier_filter_example.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+axs[1].plot(np.imag(np.fft.fft(xr_data)[1:100]))
+axs[1].plot(np.imag(np.fft.fft(no_tide)[1:100]))
+
+# Show figure
+fig.show()
+
+# Generate figure with 2 by 1 subplots and set its size (width, height) in inches
+fig, axs = plt.subplots(2, 1, dpi=100, figsize=(8, 4), constrained_layout=True)
+
+# Define start and end of data indicies
```
Comment (anissa111) [inline GeoCAT-comp-examples/matplotlib/fourier_filter_example.py:105]:
```suggestion
# Define start and end of data indices
```

Comment (anissa111) [reply GeoCAT-comp-examples/matplotlib/fourier_filter_example.py:105]: I don't know why all the comments didn't stick to the first review, sorry for the email spam!

**Comment by michaelavs** [comment]
> > @pilotchute @anissa111 When running this one locally (and attempting to build other examples with sphinx) there is a "ModuleNotFound" error from climatology.py for cf_xarray. Is this an actual naming error or is cf_xarray another package that should be added to the geocat-example conda environment?
> > Edit: actually it looks like cf_xarray isn't used in climatology after being imported, should we get rid of it in climatology?
> 
> @michaelavs Actually no! Anderson and Orhan taught me this on a recent -comp PR. You need to import cf_xarray even though the IDE will say it's not being used. I'm not getting that error locally, though.

I think it's part of the environment and that's why. Comp environment has a dependency on cf_xarray and -examples doesn't so I think thats why it isn't finding the module. I'll add it to this PR to update the environmnet.yml file so there isn't an error anymore.

**Comment by anissa111** [comment]
> I think it's part of the environment and that's why. Comp environment has a dependency on cf_xarray and -examples doesn't so I think thats why it isn't finding the module. I'll add it to this PR to update the environmnet.yml file so there isn't an error anymore.

Right, I probably added it to my local environment at some point. I agree we should probably add it to this environment, too for this example. 

As far as giving a comprehensive review, I'm not sure I know enough about what this example is supposed to do to give a good enough review. @pilotchute would you mind giving this a look when you have a minute?

**Comment by michaelavs** [comment]
@anissa111 @pilotchute So after getting the conda based packages installed, I found that the geocat-examples environment doesn't like ```from geocat.comp import fourier_filter```.  I cannot for the life of me remember how to circumvent this error with respect to the geoCAT-examples repo and/or environment. I figure I need to build something, but again, completely drawing a blank here, so whenever either of you has a moment to help me on this part, let me know!

**Review by michaelavs — APPROVED** [review]
This looks good to me! Like Anissa mentioned, it may be useful to have Austin give this a once over to make sure, but otherwise I'm happy with where it is currently.

**Review by pilotchute — APPROVED** [review]
Looks great! Thanks Erin!

**Pull request merged**